src/Tasks.js

```diff
@@ -0,0 +1,496 @@
+
+    const [taskStatus,setTaskStatus] = useState(status);
+
+    function handleChange(e) {
+        setTaskStatus(e.target.value);
+        handleStatusChange(e.target.value);
+    }
+    return (
+        <div className="taskStatusOverlay">
+            <label htmlFor="taskSatusInput"><i className="fa fa-arrow-circle-right" aria-hidden="true"></i>
+                {` Status :-`}</label>
+            <select id="taskStatusInput" value={taskStatus} onChange={handleChange}>
+                <option value="Not started">Not started</option>
+                <option value="In-progress">In-progress</option>
+                <option value="On hold">On hold</option>
+                <option value="Completed">Completed</option>
+            </select>
+        </div>
+    );
+
+}
+
+function TaskDate({date,handleDateChange}) {
+    const [taskDate,setTaskDate] = useState(date);
+
+    function handleChange(e) {
+        setTaskDate(e.target.value);
+        handleDateChange(e.target.value);
+    }
+
+    return (
+        <div className="taskDateOverlay">
+            <label htmlFor="taskDateInput"><i className="fa fa-clock-o" aria-hidden="true"></i> Due date :- </label>
+            <input value={taskDate} onChange={handleChange} type="date" id="taskDateInput" name="taskDateInput" placeholder="dd/mm/yyyy" />  
+        </div>
+    )
+}
+
+
+function TaskOverlay({task,users,handleClose,handleSave,handleDelete}) {
+
+    let selectedIndex = useMemo(() => users.findIndex((user) => user.id == task.assignee),[task]);
+    
+    let taskImage = task.imageUrl || "assets/default.png";
+    function handleImageChange(imageUrl) {
+        taskImage = imageUrl;
+    }
+
+    let taskTitle = task.title || "";
+    function handleTitleChange(title) {
+        taskTitle = title;
+    }
+
+    let taskAssignee = users[selectedIndex].name;
+    function handleAssigneeChange(assignee) {
+        taskAssignee = assignee;
+    }
+
+    let taskStatus = statusMapping[task.status];
+    function handleStatusChange(status) {
+        taskStatus = status;
+    }
+
+    let taskDate = task.dueDate || "";
+    function handleDateChange(date) {
+        taskDate = date;
+    }
+
+    const [taskStages,setTaskstages] = useState(task.stages);
+
+    function deleteStageHandle(stageName) {
+        let newStages = {};
+        for(let stage in taskStages) {
+            newStages[stage] = taskStages[stage];
+        }
+        delete newStages[stageName];
+        setTaskstages(newStages);
+    }
+
+    function addStageHandle() {
+        let stageName = prompt("Enter stage name:- ","");
+        if(stageName=="" || stageName == null)
+        {
+            return;
+        }
+        let newStages = {};
+        for(let stage in taskStages) {
+            newStages[stage] = taskStages[stage];
+        }
+        newStages[stageName] = false;
+        setTaskstages(newStages);
+    }
+
+    function closeCallback () {
+        handleClose();
```

Comment:
We don't need a separate closeCallback method? we are just invoking handleClose.